erigon-lib/seg/decompress.go

```diff
@@ -270,6 +275,13 @@ func NewDecompressor(compressedFilePath string) (d *Decompressor, err error) {
 		}
 	}
 	d.wordsStart = pos + 8 + dictSize
+
+	if d.Count() == 0 && dictSize == 0 && d.size > compressedMinSize {
+		d.Close()
```

Comment:
we return err from buildCondensedPatternTable also. maybe need add `d.Close() there also. Or close in defer with "err != nil" check.